include/openthread/legacy.h

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the top-level functions for the legacy network apis.
+ */
+
+#ifndef OPENTHREAD_LEGACY_H_
+#define OPENTHREAD_LEGACY_H_
+
+#include "openthread/platform/radio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-message
+ *
+ * @brief
+ *   This module includes functions that manipulate OpenThread message buffers.
+ *
+ * @{
+ *
+ */
+
+#define OT_LEGACY_ULA_PREFIX_LENGTH 8 ///< Legacy ULA size (in bytes)
+
+/**
+ * This method initializes the legacy processing.
+ *
+ */
+void otLegacyInit(void);
```

Comment:
@gjc13  can you please provide more background on this whole PR? Thanks.

Reply:
@gjc13 reached out to explain the reason for moving these definitions is to allow users/projects that don't go through NCP interface (i.e. want to control through OT APIs directly) can use the same legacy APIs as in NCP case.

Thinking more about this, I feel we may not want to include these in OT directly. 
In this case (where NCP interface is not being used by the user of OT i.e., project/product integrator): 
- We are declaring a bunch of new APIs for some external module to provide.
- They are not used by core of OT.
- We expect user of OT to directly call them to use them (from the app/next layer above OT).

We are unnecessarily acting as a middle-man here. No need for OT to be aware of these APIs in this case and they can be fully defined and used by the next layer and user of OT, i.e. project/product/platform related code.

In case the user wants the code to run along with OT, we provide "extension" mechanism, but in this case, I don't think user needs to add these as extension.

Regarding NCP use, we are limited in that `npc` code it the interface for users to issue commands to OT.

Thoughts?

Reply:
Correct. We can directly call those extensions in integration code.